sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -1105,14 +1105,11 @@ public void flush() {
     public Producer<Integer, Long> apply(Map<String, Object> config) {
 
       // Make sure the config is correctly set up for serializers.
-
-      // There may not be a key serializer if we're interested only in values.
-      if (config.get(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG) != null) {
-        Utils.newInstance(
-                ((Class<?>) config.get(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG))
-                        .asSubclass(Serializer.class)
-        ).configure(config, true);
-      }
+      //There should always be a key serializer. Write.values() should be setting a VoidSerializer
+      Utils.newInstance(
+              ((Class<?>) config.get(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG))
```

Comment:
How using Kafka's StringSerializer instead of VoidSerializer? We ensure that key is always null. If there is a bug in future that somehow writes non-null keys, it is better for it to fail (or write the key if the key happens to be string) rather than ignoring it here. WDYT?

Reply:
Make since. I made these changes and the latest it there.